### PR TITLE
Cleanup CDF dataclass

### LINKED ISSUE #708: Clean up remaining CDF Attrs Dataclasses Implementation
- [ ] Remove CDF Attrs Dataclasses
- [ ] Fix any remaining code that uses above classes

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary
This PR is big. I apologize in advance. If you can take a look at your instrument, I appreciate it!

## Overview
This PR cleans up CDF attribute dataclasses. This caused lot of code to fail. I tried my best to update accordingly. Unfortunately, MAG and Ultra's changes was touching lot of areas. I tried my best and added TODOs where it needs further updates. I will create GitHub issue to address those.

Following this PR, I like to refactor epoch and change other places to use 'global' epoch variable from `imap_processing/cdf/__init__.py`

### CODE REVIEW

**Comment by bourque** [comment]
The CoDICE changes LGTM! Thank you!

File: imap_processing/mag/l0/decom_mag.py
Comment (maxinelasp) [inline imap_processing/mag/l0/decom_mag.py:14]: I have a PR open to do all these changes for MAG 😢  I'm sorry I didn't merge it yet, I was prioritizing the SQS changes because I thought it would have more blockers. Since the changes in this PR would break the code, could you wait until I merge the MAG PR before merging this? I can finish it off today.

File: imap_processing/ultra/l1a/ultra_l1a.py
Comment (maxinelasp) [inline imap_processing/ultra/l1a/ultra_l1a.py]:
```suggestion

```

**Review by maxinelasp — CHANGES_REQUESTED** [review]
Can you remove the MAG changes from this review, since they're covered by my MAG PR? I can merge that first and then unblock your PR. Besides that, it looks good to me!

File: imap_processing/mag/l0/decom_mag.py
Comment (tech3371) [reply imap_processing/mag/l0/decom_mag.py:14]: oh darn..ok. I will wait for that.

File: imap_processing/cdf/config/imap_ultra_l1a_variable_attrs.yaml
Comment (laspsandoval) [inline imap_processing/cdf/config/imap_ultra_l1a_variable_attrs.yaml:2]: Noted.

**Review by laspsandoval — COMMENTED** [review]
It looks good. Make certain tests pass for Ultra. Thanks for doing this.

File: imap_processing/cdf/config/imap_ultra_l1a_variable_attrs.yaml
Comment (tech3371) [reply imap_processing/cdf/config/imap_ultra_l1a_variable_attrs.yaml:2]: Thank you, Laura! I created this issue which has more details about which files need further updates. https://github.com/IMAP-Science-Operations-Center/imap_processing/issues/716. Hope that helps

File: imap_processing/mag/l0/decom_mag.py
Comment (maxinelasp) [reply imap_processing/mag/l0/decom_mag.py:14]: Sorry for the confusion!

Comment (tech3371) [reply imap_processing/mag/l0/decom_mag.py:14]: No worries. I totally forgot that that PR is open.

File: imap_processing/cdf/config/imap_constant_attrs.yaml
Comment (tech3371) [inline imap_processing/cdf/config/imap_constant_attrs.yaml:23]: This may go away when I switch function that was using `common_cdf_attrs.py` to new function `packet_file_to_datasets()`

**Comment by sdhoyt** [comment]
Lo and HIT look good. Thanks!

File: imap_processing/tests/cdf/test_imap_cdf_manager.py
Comment (tmplummer) [inline imap_processing/tests/cdf/test_imap_cdf_manager.py]: This test is run with the non-standard source directory. That means that the new call to `self.load_variable_attributes("imap_constant_attrs.yaml") is failing because it looks for that file in `imap_processing/tests/cdf`. Since this is just testing coverage for manually setting the source dir, I would suggest that mocking out the `load_variable_attributes` function is appropriate.

**Review by tmplummer — APPROVED** [review]
Hi looks good. Thanks you!

File: imap_processing/tests/cdf/test_imap_cdf_manager.py
Comment (tech3371) [reply imap_processing/tests/cdf/test_imap_cdf_manager.py]: Thank you for that tip! That fixed that test fail.

**Review by maxinelasp — APPROVED** [review]
Fixed issues, sounds good

**Pull request merged**
